ale_linters/php/psalm.vim

```diff
@@ -0,0 +1,38 @@
+" Author: richard marmorstein <https://github.com/twitchard>
+" Description: plugin for Psalm, static analyzer for PHP
+
+let g:ale_php_psalm_executable = get(g:, 'ale_php_psalm_executable', 'psalm')
+
+function! ale_linters#php#psalm#GetExecutable(buffer) abort
+    return ale#Var(a:buffer, 'php_psalm_executable')
+endfunction
+
+function! ale_linters#php#psalm#GetCommand(buffer) abort
+    return ale#Escape(ale_linters#php#psalm#GetExecutable(a:buffer)) . ' --diff --output-format=emacs %s'
```

Comment:
You can write `command: '%e --diff --output-format=emacs %s`' below instead, and get the same result.